### PR TITLE
37 parse error fix for hapmap and  gt matrix formats

### PR DESCRIPTION
**Pull request by Jiu9Shen**
## Metadata

 - Issue #

- [ ] I feel this PR is ready to be merged.
- [x] This PR is dependent upon [PR #36 ]

Documentation:
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Testing?

- [ ] I tested on a generic Tripal Site
- [x] I tested on a KnowPulse Clone
- [ ] This PR includes automated testing

### CODE REVIEW

**Comment by laceysanderson** [comment]
Dependent PR merged. Just waiting on second review (from @carolyncaron )

**Comment by laceysanderson** [comment]
Updated from master to make the code changes more clear :-)

**Comment by Jiu9Shen** [comment]
Sorry, that was a mistake.

**Comment by carolyncaron** [comment]
I was going to suggest making this code an API function, when I noticed that vcf_filter_read_VCF_into_array already solves this issue. It seems this function already pulls out the GT field regardless of its position. Additionally, it has the added benefit of using the PL field for verifying the call. It would probably help maintainability a lot to use already pre-existing code!

File: includes/vcf_filter.trpdownload.genotype_matrix.inc [hunk 1]
```diff
@@ -152,17 +152,43 @@ function vcf_filter_convert_VCF_to_GenotypeMatrix($in_vcf_filename, $out_Genotyp
...
 
+      // need to check 8th column to determine the postiion of GT
+      // the format in 8th column should look like:
+      //   GT:GQ:DP:HQ (VCF specification can be found in https://samtools.github.io/hts-specs/VCFv4.2.pdf)
+      // normally, GT should be the first element, but it not alway the case, but GT not exist is not considered possible here
+      // there is possibiliby of ONLY GT available
+      // set default GT position to be 0, the first element
+      $gt_position = 0;
+      if (strlen($splitline[8]) > 2){
+        $cvf_8c_format = explode(":", $splitline[8]);
```
Comment (carolyncaron) [inline includes/vcf_filter.trpdownload.genotype_matrix.inc]: cvf should be vcf?

Comment (Jiu9Shen) [reply includes/vcf_filter.trpdownload.genotype_matrix.inc]: fixedbitmoji

**Comment by Jiu9Shen** [comment]
@carolyncaron are you suggesting to use function vcf_filter_read_VCF_into_array() for formats hapmap and phenotype_matrix?

**Pull request merged**